src/model/neural_network/activation_function_utils.py

```diff
@@ -0,0 +1,20 @@
+from collections import namedtuple
+
+from model.neural_network.activation_functions import (
+    sigmoid_derivative_from_value,
+    relu,
+    relu_derivative_from_value,
+    tanh_derivative_from_value,
+    sigmoid,
+    tanh,
+)
+
+ActivationFunctionUtils = namedtuple(
+    "ActivationFunctionUtils", ["function", "derivative"]
+)
```

Comment:
This can be a dataclass, which works better with mypy. The name is also not very descriptive.
```suggestion
@dataclass
class ActivationFunction:
    function: Callable[[np.ndarray], np.ndarray]
    derivative_from_value: Callable[[np.ndarray], np.ndarray]
```